addons/sale_timesheet/models/account.py

```diff
@@ -71,9 +71,11 @@ def _check_can_write(self, values):
     def _timesheet_preprocess(self, values):
         values = super(AccountAnalyticLine, self)._timesheet_preprocess(values)
         # task implies so line (at create)
-        if 'task_id' in values and not values.get('so_line') and values.get('employee_id'):
+        if 'task_id' in values and not values.get('so_line') and (values.get('employee_id') or len(self.mapped('employee_id')) > 0):
+            if not values.get('employee_id') and len(self.mapped('employee_id')) > 1:
+                raise UserError(_('You can not modify timesheets from different employees'))
             task = self.env['project.task'].sudo().browse(values['task_id'])
-            employee = self.env['hr.employee'].sudo().browse(values['employee_id'])
+            employee = self.env['hr.employee'].sudo().browse(values['employee_id']) if values.get('employee_id') else self.mapped('employee_id')
```

Comment:
```suggestion
            employee = self.env['hr.employee'].sudo().browse(values['employee_id']) if values.get('employee_id') else self.employee_id
```

Reply:
As you said before, self can be a recordset and self.employee_id would crash.
The user error is only raise if there are multiples different employee_id.

Reply:
indeed, I missed that